pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -268,6 +234,50 @@ func desiredDNSConfigMap(dns *operatorv1.DNS, clusterDomain string, caBundleRevi
 	return cm, nil
 }
 
+// validateTLSSettings validates if serverName is set when tls is enabled, and validates if tls is configured only for ip addresses.
+// If a serverName is set, but tls is not enabled, then the serverName is ignored.
+func validateTLSSettings(dns *operatorv1.DNS) (*operatorv1.DNS, error) {
```

Comment:
`validateTLSSettings` actually returns a modified version of its input, so it's more like `sanitizeTLSSettings`; its name and godoc should describe what it does.

Reply:
Done.